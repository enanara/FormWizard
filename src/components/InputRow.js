import React from 'react';
import './InputRow.css';

const InputRow = ({labelText, className, callback, inputText}) => {
    const onType = (event) => {
        callback(event.target.value);
    }
    return (
        <div className = {`input-row ${className}`}>
            <p>
                {labelText}
            </p>
            <input
                data-testid = 'input-row-input'
                value = {inputText}
                onChange = {onType}
            />
        </div>
    )
}

export default InputRow;