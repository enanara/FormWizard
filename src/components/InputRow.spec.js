import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import InputRow from './InputRow';

const renderInputRow = (props) => {
    return render(<InputRow {...props}/>);
}

describe('InputRow', () => {
    it('should exist ', () => {
        const {container} = renderInputRow();
        expect(container).toMatchSnapshot();
    });
    it('should call callback on change', () => {
        const callback = jest.fn();
        
        const {getByTestId} = renderInputRow({
            labelText: 'testing', 
            className:'input-row-testing', 
            callback, 
            inputText: '',
        });
        fireEvent.change(getByTestId('input-row-input'), {target: {value: 'bunnyRabbit'}});
        expect(callback).toHaveBeenCalledWith('bunnyRabbit');
    });
})