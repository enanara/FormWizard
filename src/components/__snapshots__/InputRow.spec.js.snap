// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRow should exist  1`] = `
<div>
  <div
    class="input-row undefined"
  >
    <p />
    <input
      data-testid="input-row-input"
      value=""
    />
  </div>
</div>
`;
